ros/src/i2c_com/scripts/BNO055.py

```diff
@@ -0,0 +1,180 @@
+                'roll':  0,  # Rotation about y axix (perpindicular to the pins IMU) +/- 0.01 degree
+                'pitch': 0,  # Rotation about x axis (parallel to the pins of IMU) +/- 0.01 degree
+
+            },
+            'gyro': {
+                'x': 0, # 3e-2 degree/sec
+                'y': 0, # 3e-2 degree/sec
+                'z': 0, # 3e-2 degree/sec
+            },
+            'acceleration': {
+                'x': 0, # +/- 5e-4 g
+                'y': 0, # +/- 5e-4 g
+                'z': 0, # +/- 5e-4 g
+            },
+            'linear_acceleration': {
+                'x': 0, # +/- 0.25 m/s^2
+                'y': 0, # +/- 0.25 m/s^2
+                'z': 0, # +/- 0.25 m/s^2
+            },
+            'temp': 0, # Good enough
+        }
+
+    @property
+    def data(self):
+        return self._data
+
+    def roll(self):
+        return self._data['euler']['roll']
+
+    def pitch(self):
+        return self._data['euler']['pitch']
+
+    def yaw(self):
+        return self._data['euler']['yaw']
+
+    def gyro_x(self):
+        return self._data['gyro']['x']
+    def gyro_y(self):
+        return self._data['gyro']['y']
+    def gyro_z(self):
+        return self._data['gyro']['z']
+
+    def acceleration_x(self):
+        return self._data['acceleration']['x']
+    def acceleration_y(self):
+        return self._data['acceleration']['y']
+    def acceleration_z(self):
+        return self._data['acceleration']['z']
+
+    def linear_acceleration_x(self):
+        return self._data['linear_acceleration']['x']
+    def linear_acceleration_y(self):
+        return self._data['linear_acceleration']['y']
+    def linear_acceleration_z(self):
+        return self._data['linear_acceleration']['z']
+
+
+    def update(self):
+        euler = self._bno.read_euler()
+        self._data['euler']['yaw']     = euler[0]
+        self._data['euler']['roll']    = euler[1]
+        self._data['euler']['pitch']   = euler[2]
+
+	gyro = self._bno.read_gyroscope()
+        self._data['gyro']['x'] = gyro[0]
+        self._data['gyro']['y'] = gyro[1]
+        self._data['gyro']['z'] = gyro[2]
+
+        acceleration = self._bno.read_accelerometer()
+        self._data['acceleration']['x'] = acceleration[0]
+        self._data['acceleration']['y'] = acceleration[1]
+        self._data['acceleration']['z'] = acceleration[2]
+
+        linear_accel = self._bno.read_linear_acceleration()
+        self._data['linear_acceleration']['x'] = linear_accel[0]
+        self._data['linear_acceleration']['y'] = linear_accel[1]
+        self._data['linear_acceleration']['z'] = linear_accel[2]
+
+        temp = self._bno.read_temp()
+        self._data['temp'] = temp
+
+        return True
+
+    def get_calibration(self):
+        return self._bno.get_calibration()
+
+    def reset_calibration(self):
+        cal_array_original = self.get_calibration()
+        self._bno.set_calibration(self._bno.get_calibration())
```

Comment:
Why call get_calibration() again? Just pass cal_array_original right?

Reply:
the get calibration call is to the adafruit library. It is calling the constant hard coded and defined in there. This code is for if we do custom calibration or the calibration data is corrupted we want to get a log of what the calibration data was before we reset it to the library standard